### PR TITLE
speed-up deployer by running in parallel processes

### PR DESCRIPTION
**Pull request by OriHoch**

### CODE REVIEW

File: cwm_worker_operator/deployer.py [hunk 1]
```diff
@@ -217,12 +222,30 @@ def get_deployment_config(debug, domains_config, extra_minio_extra_configs, name
...
 
-def run_single_iteration(domains_config: domains_config_module.DomainsConfig, metrics, deployments_manager, extra_minio_extra_configs=None, **_):
-    deployer_metrics = metrics
+class DeployerMultiprocessor(Multiprocessor):
+
+    def _run_async(self, worker_id, extra_minio_extra_configs, domains_config, metrics, deployments_manager, flow_manager):
+        cmd = ['cwm-worker-operator', 'deployer', 'deploy_worker', '--worker-id', worker_id]
+        if extra_minio_extra_configs:
+            cmd += ['--extra-minio-extra-configs', json.dumps(extra_minio_extra_configs)]
+        return subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```
Comment (iamazeem) [inline cwm_worker_operator/deployer.py:231]: Shouldn't the `Multiprocessor` itself take the responsibility of creating the `subprocess.Popen`?

File: cwm_worker_operator/multiprocessor.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    def wait_for_all_processes_to_complete(self):
+        while self.get_active_processes() != 0:
+            time.sleep(1)
+
+    def process(self, *args, **kwargs):
+        if self.is_async:
+            key = self._get_process_key(*args, **kwargs)
+            assert key not in self.processes
+            self.processes[key] = self._run_async(*args, **kwargs)
```
Comment (iamazeem) [inline cwm_worker_operator/multiprocessor.py:37]: How can `_run_async` here enforce that it should always return a process?

**Review by iamazeem — APPROVED** [review]
LGTM! :+1: It's a nice optimization! :trophy:

File: cwm_worker_operator/deployer.py (hunk 1 above)
Comment (OriHoch) [reply cwm_worker_operator/deployer.py:231]: I want to provide flexibility in creating it, maybe other processes will want to create if differently, do other actions before / after creating the process

File: cwm_worker_operator/multiprocessor.py (hunk 2 above)
Comment (OriHoch) [reply cwm_worker_operator/multiprocessor.py:37]: could check that it's a Popen class using `isinstance` but I prefer not to have too many checks and rely on it raising an Exception if something fails..

**Pull request merged**